### PR TITLE
[Update] use html to improve table style

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Comment by Oreoxmt** [comment]
The preview link is [here](https://preview-dev-guide.pingcap-docs-website-scaffold.pages.dev/zh/tidb/dev/dev-guide-third-party-support).

File: develop/dev-guide-third-party-support.md [hunk 1]
```diff
@@ -32,33 +32,153 @@ PingCAP 与开源社区合作，通过三方工具提供以下支持：
 | C#(.Net) | [MySQL Connector/NET](https://downloads.mysql.com/archives/c-net/) | 8.0 | Compatible | N/A | N/A |
 | ODBC | [MySQL Connector/ODBC](https://downloads.mysql.com/archives/c-odbc/) | 8.0 | Compatible | N/A | N/A |
 | Go | [go-sql-driver/mysql](https://github.com/go-sql-driver/mysql) | v1.6.0 | Full | N/A | [TiDB 和 Golang 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-golang.md) |
-| Java | [JDBC](https://dev.mysql.com/downloads/connector/j/) | 8.0 | Full | [pingcap/mysql-connector-j](/develop/dev-guide-choose-driver-or-orm.md#java-drivers) <br/>[pingcap/tidb-loadbalance](/develop/dev-guide-choose-driver-or-orm.md#tidb-loadbalance) | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md) |
+| Java | [JDBC](https://dev.mysql.com/downloads/connector/j/) | 8.0 | Full | [pingcap/mysql-connector-j](/develop/dev-guide-choose-driver-or-orm.md#java-drivers) <hr style="margin: 5px 0; height: 0.1em"/> [pingcap/tidb-loadbalance](/develop/dev-guide-choose-driver-or-orm.md#tidb-loadbalance) | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md) |
```
Comment (Mini256) [inline develop/dev-guide-third-party-support.md:35]: Maybe use `<ul>` and `<li>` to list the drivers will be better.

File: develop/dev-guide-third-party-support.md [hunk 2]
```diff
@@ -32,33 +32,153 @@ PingCAP 与开源社区合作，通过三方工具提供以下支持：
...
+         <th>TiDB 适配器</th>
+         <th>教程</th>
+      </tr>
+   </thead>
+   <tbody>
+      <tr>
+         <td rowspan="4">Go</td>
+         <td><a href="https://github.com/go-gorm/gorm" target="_blank" referrerpolicy="no-referrer-when-downgrade">gorm</a></td>
+         <td>v1.23.5</td>
+         <td rowspan="4">Full</td>
```
Comment (Mini256) [inline develop/dev-guide-third-party-support.md:58]: I think we don't need to enable rowspan for the `support level` column. Just keep the structure of the table simple.

**Comment by TomShawn** [comment]
@shczhen PTAL whether the docs site supports the HTML table syntax. Thanks~

File: develop/dev-guide-third-party-support.md (hunk 2 above)
Comment (Icemap) [reply develop/dev-guide-third-party-support.md]: Got it.

**Review by shczhen — APPROVED** [review]

**Comment by shczhen** [comment]
/lgtm

image
image
image

**Comment by Icemap** [comment]
@TomShawn  PTAL

**Comment by TomShawn** [comment]
> @TomShawn PTAL

LGTM. Great work!

**Review by Mini256 — APPROVED** [review]

**Pull request merged**